backend/arm64/proc.ml

```diff
@@ -146,15 +144,17 @@ let hard_int_reg =
   done;
   v
 
-let hard_float_reg =
+let hard_float_reg_gen kind =
   let v = Array.make 32 Reg.dummy in
   for i = 0 to 31 do
-    v.(i) <- Reg.at_location Float (Reg(100 + i))
+    v.(i) <- Reg.at_location kind (Reg(100 + i))
```

Comment:
Shouldn't `100` depend on the kind,
or are they synonyms?

Reply:
I've refactored it slightly, see what you think. 

Constants like this `100` are used about a hunder times all over the backend as you know and we should clean it up separately. In this particular place, `hard_float_reg_gen` was only used with machtype that falls into the `register_class` of float, as the name indicates. I can generalize it slightly for `hard_int_reg` and avoid magic constants.

Reply:
Yes, that should go when we refactor
the classes indeed. You are right, we
should probably not spend more time
on it right now.